Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -103,33 +94,34 @@ public class PythonLanguageClient : ILanguageClient, ILanguageClientCustomMessag
                 return null;
             }
 
-            if (PythonWorkspaceContextProvider.Workspace != null) {
-                _clientContext = new PythonLanguageClientContextWorkspace(PythonWorkspaceContextProvider.Workspace, PythonCoreConstants.ContentType);
-            } else {
-                _clientContext = new PythonLanguageClientContextGlobal(OptionsService, PythonCoreConstants.ContentType);
-            }
+            _clientContext = PythonWorkspaceContextProvider.Workspace != null 
+                ? (IPythonLanguageClientContext)new PythonLanguageClientContextWorkspace(PythonWorkspaceContextProvider.Workspace, PythonCoreConstants.ContentType) 
+                : new PythonLanguageClientContextGlobal(OptionsService, PythonCoreConstants.ContentType);
+
+            _clientContext.InterpreterChanged += OnInterpreterChanged;
+            _clientContext.SearchPathsChanged += OnSearchPathsChanged;
+            _disposables.Add(() => {
+                _clientContext.InterpreterChanged -= OnInterpreterChanged;
+                _clientContext.SearchPathsChanged -= OnSearchPathsChanged;
```

Comment:
I've noticed that we don't have anyone disposing `_clientContext`, I think it should be added here after the event unregistration.

Reply:
Yep, added disposing.